### ISSUE TITLE
Provide method to generate output for HdrHistogram Plotter

### ISSUE THREAD

**Issue by mpeterson0x0** [HdrHistogram/HdrHistogram_py#4]
The Java version of HdrHistogram has an `outputPercentileDistribution` method that generates the output required by the HdrHistogram Plotter here:  http://hdrhistogram.github.io/HdrHistogram/plotFiles.html

I reviewed the HdrHistogram_py codebase and could not find any similar functionality. 

If I missed it, could you provide documentation on how to call it?

If it is not yet present, please consider this a request to add that functionality.

Thank you!

**Comment by ahothan** [comment]
This function has not been ported yet (as it was not needed) but it can certainly be added.
If there is any volunteer for this work I'd be happy to pull the changes, otherwise I'll need to find time to add it. 
Would be helpful to have the output of the Java test code that calls that API if you have that.

https://github.com/HdrHistogram/HdrHistogram/blob/2413b056441604ee2a41dbb546fd91f9ee50d9be/src/examples/java/SimpleHistogramExample.java

**Comment by mpeterson0x0** [comment]
The output from the Java `histogram.outputPercentileDistribution(System.out, 1000.0)` call in `SimpleHistogramExample` is:

```
       Value     Percentile TotalCount 1/(1-Percentile)

       2.787 0.000000000000          3           1.00
       2.869 0.100000000000     483641           1.11
       2.885 0.200000000000    1017428           1.25
       2.897 0.300000000000    1508961           1.43
       2.907 0.400000000000    1907091           1.67
       2.921 0.500000000000    2389592           2.00
       2.929 0.550000000000    2621985           2.22
       2.939 0.600000000000    2866237           2.50
       2.951 0.650000000000    3107793           2.86
       2.965 0.700000000000    3337906           3.33
       2.981 0.750000000000    3561791           4.00
       2.991 0.775000000000    3674700           4.44
       3.003 0.800000000000    3792817           5.00
       3.017 0.825000000000    3914461           5.71
       3.035 0.850000000000    4031373           6.67
       3.059 0.875000000000    4153674           8.00
       3.075 0.887500000000    4212488           8.89
       3.093 0.900000000000    4269216          10.00
       3.119 0.912500000000    4327088          11.43
       3.155 0.925000000000    4386474          13.33
       3.209 0.937500000000    4444485          16.00
       3.253 0.943750000000    4473435          17.78
       3.303 0.950000000000    4503306          20.00
       3.349 0.956250000000    4532788          22.86
       3.409 0.962500000000    4562266          26.67
       3.479 0.968750000000    4592260          32.00
       3.509 0.971875000000    4606540          35.56
       3.539 0.975000000000    4621411          40.00
       3.569 0.978125000000    4636121          45.71
       3.601 0.981250000000    4650775          53.33
       3.637 0.984375000000    4665405          64.00
       3.659 0.985937500000    4673027          71.11
       3.685 0.987500000000    4680464          80.00
       3.717 0.989062500000    4687850          91.43
       3.761 0.990625000000    4694930         106.67
       3.839 0.992187500000    4702302         128.00
       3.909 0.992968750000    4706025         142.22
       4.071 0.993750000000    4709723         160.00
       4.395 0.994531250000    4713404         182.86
       4.699 0.995312500000    4717104         213.33
       4.931 0.996093750000    4720830         256.00
       5.039 0.996484375000    4722709         284.44
       5.159 0.996875000000    4724543         320.00
       5.315 0.997265625000    4726379         365.71
       5.539 0.997656250000    4728218         426.67
       6.227 0.998046875000    4730064         512.00
      11.343 0.998242187500    4730988         568.89
      12.039 0.998437500000    4731921         640.00
      12.215 0.998632812500    4732879         731.43
      12.423 0.998828125000    4733788         853.33
      12.983 0.999023437500    4734707        1024.00
      13.135 0.999121093750    4735156        1137.78
      13.239 0.999218750000    4735639        1280.00
      13.335 0.999316406250    4736109        1462.86
      13.455 0.999414062500    4736555        1706.67
      13.631 0.999511718750    4737006        2048.00
      13.759 0.999560546875    4737243        2275.56
      13.911 0.999609375000    4737472        2560.00
      14.063 0.999658203125    4737702        2925.71
      14.263 0.999707031250    4737931        3413.33
      14.543 0.999755859375    4738167        4096.00
      14.727 0.999780273438    4738279        4551.11
      14.983 0.999804687500    4738395        5120.00
      15.279 0.999829101563    4738511        5851.43
      15.519 0.999853515625    4738628        6826.67
      15.727 0.999877929688    4738741        8192.00
      15.863 0.999890136719    4738799        9102.22
      16.111 0.999902343750    4738857       10240.00
      16.543 0.999914550781    4738914       11702.86
      17.023 0.999926757813    4738971       13653.33
      17.551 0.999938964844    4739030       16384.00
      17.695 0.999945068359    4739058       18204.44
      17.919 0.999951171875    4739087       20480.00
      18.239 0.999957275391    4739116       23405.71
      18.783 0.999963378906    4739145       27306.67
      19.679 0.999969482422    4739174       32768.00
      20.383 0.999972534180    4739188       36408.89
      22.031 0.999975585938    4739203       40960.00
      91.263 0.999978637695    4739217       46811.43
      91.967 0.999981689453    4739233       54613.33
      92.287 0.999984741211    4739248       65536.00
      92.415 0.999986267090    4739254       72817.78
      92.607 0.999987792969    4739262       81920.00
      92.799 0.999989318848    4739268       93622.86
      93.119 0.999990844727    4739276      109226.67
      93.375 0.999992370605    4739282      131072.00
      93.503 0.999993133545    4739287      145635.56
      93.631 0.999993896484    4739291      163840.00
      93.695 0.999994659424    4739293      187245.71
      94.207 0.999995422363    4739297      218453.33
      94.463 0.999996185303    4739300      262144.00
      94.719 0.999996566772    4739302      291271.11
      94.911 0.999996948242    4739305      327680.00
      95.103 0.999997329712    4739306      374491.43
      95.231 0.999997711182    4739308      436906.67
      95.615 0.999998092651    4739309      524288.00
      95.871 0.999998283386    4739311      582542.22
      95.871 0.999998474121    4739311      655360.00
      95.935 0.999998664856    4739312      748982.86
      96.127 0.999998855591    4739313      873813.33
     100.927 0.999999046326    4739314     1048576.00
     100.927 0.999999141693    4739314     1165084.44
     105.471 0.999999237061    4739315     1310720.00
     105.471 0.999999332428    4739315     1497965.71
  106627.071 0.999999427795    4739316     1747626.67
  106627.071 0.999999523163    4739316     2097152.00
  106627.071 0.999999570847    4739316     2330168.89
  222298.111 0.999999618530    4739317     2621440.00
  222298.111 0.999999666214    4739317     2995931.43
  222298.111 0.999999713898    4739317     3495253.33
  222298.111 0.999999761581    4739317     4194304.00
  222298.111 0.999999785423    4739317     4660337.78
  230293.503 0.999999809265    4739318     5242880.00
  230293.503 1.000000000000    4739318
#[Mean    =        3.112, StdDeviation   =      154.926]
#[Max     =   230293.503, Total count    =      4739318]
#[Buckets =           32, SubBuckets     =         2048]
```

**Comment by ahothan** [comment]
Closing this case as this has been addressed by latest commit (0.5.0 tag).
Many thanks for @lebinh for his contribution!
For usage, check for the test code test_output_percentile_distribution()
in https://github.com/HdrHistogram/HdrHistogram_py/blob/master/test/test_hdrhistogram.py

Output of the new histogram method output_percentile_distribution  to print a histogram:

```
       Value     Percentile TotalCount 1/(1-Percentile)

       1.000 0.000000000000      10000           1.00
       1.000 0.100000000000      10000           1.11
       1.000 0.190000000000      10000           1.23
       1.000 0.271000000000      10000           1.37
       1.000 0.343900000000      10000           1.52
       1.000 0.409510000000      10000           1.69
       1.000 0.468559000000      10000           1.88
       1.000 0.521703100000      10000           2.09
       1.000 0.569532790000      10000           2.32
       1.000 0.612579511000      10000           2.58
       1.000 0.651321559900      10000           2.87
       1.000 0.686189403910      10000           3.19
       1.000 0.717570463519      10000           3.54
       1.000 0.745813417167      10000           3.93
       1.000 0.771232075450      10000           4.37
       1.000 0.794108867905      10000           4.86
       1.000 0.814697981115      10000           5.40
       1.000 0.833228183003      10000           6.00
       1.000 0.849905364703      10000           6.66
       1.000 0.864914828233      10000           7.40
       1.000 0.878423345409      10000           8.23
       1.000 0.890581010868      10000           9.14
       1.000 0.901522909782      10000          10.15
       1.000 0.911370618803      10000          11.28
       1.000 0.920233556923      10000          12.54
       1.000 0.928210201231      10000          13.93
       1.000 0.935389181108      10000          15.48
       1.000 0.941850262997      10000          17.20
       1.000 0.947665236697      10000          19.11
       1.000 0.952898713028      10000          21.23
       1.000 0.957608841725      10000          23.59
       1.000 0.961847957552      10000          26.21
       1.000 0.965663161797      10000          29.12
       1.000 0.969096845617      10000          32.36
       1.000 0.972187161056      10000          35.95
       1.000 0.974968444950      10000          39.95
       1.000 0.977471600455      10000          44.39
       1.000 0.979724440410      10000          49.32
       1.000 0.981751996369      10000          54.80
       1.000 0.983576796732      10000          60.89
       1.000 0.985219117059      10000          67.65
       1.000 0.986697205353      10000          75.17
       1.000 0.988027484817      10000          83.52
       1.000 0.989224736336      10000          92.81
       1.000 0.990302262702      10000         103.12
       1.000 0.991272036432      10000         114.57
       1.000 0.992144832789      10000         127.30
       1.000 0.992930349510      10000         141.45
       1.000 0.993637314559      10000         157.17
       1.000 0.994273583103      10000         174.63
       1.000 0.994846224793      10000         194.03
       1.000 0.995361602313      10000         215.59
       1.000 0.995825442082      10000         239.55
       1.000 0.996242897874      10000         266.16
       1.000 0.996618608086      10000         295.74
       1.000 0.996956747278      10000         328.60
       1.000 0.997261072550      10000         365.11
       1.000 0.997534965295      10000         405.67
       1.000 0.997781468766      10000         450.75
       1.000 0.998003321889      10000         500.83
       1.000 0.998202989700      10000         556.48
       1.000 0.998382690730      10000         618.31
       1.000 0.998544421657      10000         687.01
       1.000 0.998689979491      10000         763.35
       1.000 0.998820981542      10000         848.16
       1.000 0.998938883388      10000         942.40
       1.000 0.999044995049      10000        1047.11
       1.000 0.999140495544      10000        1163.46
       1.000 0.999226445990      10000        1292.73
       1.000 0.999303801391      10000        1436.37
       1.000 0.999373421252      10000        1595.97
       1.000 0.999436079127      10000        1773.30
       1.000 0.999492471214      10000        1970.33
       1.000 0.999543224093      10000        2189.26
       1.000 0.999588901683      10000        2432.51
       1.000 0.999630011515      10000        2702.79
       1.000 0.999667010363      10000        3003.10
       1.000 0.999700309327      10000        3336.77
       1.000 0.999730278394      10000        3707.53
       1.000 0.999757250555      10000        4119.47
       1.000 0.999781525499      10000        4577.19
       1.000 0.999803372950      10000        5085.77
       1.000 0.999823035655      10000        5650.86
       1.000 0.999840732089      10000        6278.73
       1.000 0.999856658880      10000        6976.37
       1.000 0.999870992992      10000        7751.52
       1.000 0.999883893693      10000        8612.80
       1.000 0.999895504324      10000        9569.77
  100007.935 0.999905953891      10001       10633.08
  100007.935 1.000000000000      10001
#[Mean    =       10.996, StdDeviation   =   999641.716]
#[Max     =   100007.935, TotalCount     =    10001.000]
#[Buckets =           22, SubBuckets     =         2048]
```
